normandy/recipes/tests/__init__.py

```diff
@@ -21,6 +23,10 @@ class Meta:
     name = FuzzyUnicode()
     implementation = 'console.log("test");'
 
+    @factory.lazy_attribute
+    def implementation_hash(action):
+        return hashlib.sha1(action.implementation.encode()).hexdigest()
```

Comment:
Factories don't yet have a Django model when they are being built (when this is called), so I duplicated this logic from the model. There is a test in `test_factories.py` to make sure this stays in sync.